manager/kubernetes.go

```diff
@@ -115,3 +116,60 @@ func (m *VolumeManager) PVCCreate(name, namespace, pvcName string) (v *longhorn.
 func (m *VolumeManager) GetDaemonSetRO(name string) (*appsv1.DaemonSet, error) {
 	return m.ds.GetDaemonSet(name)
 }
+
+func (m *VolumeManager) PVCExpand(volumeName string, size string) (v *longhorn.Volume, err error) {
+	defer func() {
+		err = errors.Wrapf(err, "unable to expand volume %v", volumeName)
+	}()
+
+	v, err = m.ds.GetVolume(volumeName)
+	if err != nil {
+		return nil, err
+	}
+
+	if v.Status.State != types.VolumeStateDetached {
+		return nil, fmt.Errorf("invalid volume state to expand: %v", v.Status.State)
+	}
+
+	if types.GetCondition(v.Status.Conditions, types.VolumeConditionTypeScheduled).Status != types.ConditionStatusTrue {
+		return nil, fmt.Errorf("cannot expand volume before replica scheduling success")
+	}
+
+	pvcName := v.Status.KubernetesStatus.PVCName
+	resourceSize := resource.MustParse(size)
+	if pvcName != "" {
+		pvcNamespace := v.Status.KubernetesStatus.Namespace
+		pvc, err := m.ds.GetPersisentVolumeClaim(pvcNamespace, pvcName)
+		if err != nil {
+			return nil, err
+		}
+
+		if pvc.Status.Capacity.StorageEphemeral().Cmp(resourceSize) > 0 {
```

Comment:
I don't think this is using the right field? you probably want to check against the requested size instead?

Reply:
I was intended to use `pvc.Status.Capacity.Storage()`, but the API is only support since v0.18.x.  Currently,  `longhorn-manager` and `longhorn-share-manager` has specified version to use v0.16.15.  Do you know the reason why we didn't update it?  Should we update it?

Reply:
Seems like it is due to https://github.com/longhorn/longhorn/issues/1938, our minimum Kubernetes requirements is 1.16+.  Without the API, there is no reason to check it.